test/jest/__tests__/impl/install_logic/Installer.Tests.spec.ts

```diff
@@ -122,26 +92,81 @@ describe('Installer Tests', () => {
 
     });
 
+    describe('SUBDIR_NO_FLATTEN', () => {
+        beforeEach(async () => {
+            await installLogicBeforeEach('RiskOfRain2');
+        });
+
+        test('Copy to root', async () => {
+            InstallationRules.RULES = [{
+                gameName: 'RiskOfRain2',
+                rules: [{
+                    route: '.',
+                    trackingMethod: 'SUBDIR_NO_FLATTEN',
+                    subRoutes: [],
+                    defaultFileExtensions: [],
+                    isDefaultLocation: true
+                }],
+                relativeFileExclusions: null,
+            }];
+
+            const pkg = createManifest('test_mod', 'auth');
+            const cachePkgRoot = path.join(PathResolver.MOD_ROOT, 'cache', pkg.getName(), pkg.getVersionNumber().toString());
+            await FsProvider.instance.mkdirs(path.join(cachePkgRoot, 'folder', 'subfolder'));
+            await FsProvider.instance.writeFile(path.join(cachePkgRoot, 'folder', 'subfolder', 'loose.file'), '');
+            await FsProvider.instance.writeFile(path.join(cachePkgRoot, 'manifest.json'), '');
+
+            const profile = Profile.getActiveProfile().asImmutableProfile();
+            await ProfileInstallerProvider.instance.installMod(pkg, profile);
+
+            expect(await FsProvider.instance.exists(
+                profile.joinToProfilePath(pkg.getName(), 'folder', 'subfolder', 'loose.file')
+            )).toBeTruthy();
+
+            expect(await FsProvider.instance.exists(
+                profile.joinToProfilePath(pkg.getName(), 'manifest.json')
+            )).toBeTruthy();
+        });
+
+        test('No default location', async () => {
```

Comment:
Is this the behaviour we'd expect? What would we expect to happen in this case - Maybe an error to be thrown?

Happy if this is refactoring to existing behaviour, not a blocker for this PR and can revisit what we think should happen.

Reply:
This is just adding game-agnostic tests for the tracking method that didn't have any before. The actual implementation of the tracking method hasn't changed. How this (and other) tracking methods should work in special cases or misconfigurations is an interesting question but I don't have answers to it.